### ISSUE TITLE
KotlinGenerator generates invalid code in equals() and hashCode() methods for inline value class fields

### ISSUE THREAD

**Issue by apatrida** [jOOQ/jOOQ#14564]
### Expected behavior

in the CVE project, I changed to 3.18.0 JOOQ / Kotlin 1.8.0 and added these values to maven codegen config:

```
<generatedAnnotation>false</generatedAnnotation>
<interfaces>true</interfaces>
<pojos>true</pojos>
<immutableInterfaces>true</immutableInterfaces>

<PojosAsKotlinDataClasses>true</PojosAsKotlinDataClasses>
<kotlinNotNullPojoAttributes>true</kotlinNotNullPojoAttributes>
<kotlinNotNullRecordAttributes>true</kotlinNotNullRecordAttributes>
<kotlinNotNullInterfaceAttributes>true</kotlinNotNullInterfaceAttributes>
```

this caused the record class `.equals()` method to have this code:

```  
if (this.id === null) {
            if (o.id !== null)
                return false
}
```

which does not compile:

```
Identity equality for arguments of types TestId? and Nothing? is forbidden
```

This happens when the `id` column is converted into an inline value class and is not the raw Int as previous.

### Steps to reproduce the problem

broken	

### jOOQ Version

3.18.0-SNAPSHOT

### Database product and version

H2

**Comment by apatrida** [comment]
[jOOQ-mcve-kotlin-14564.zip](https://github.com/jOOQ/jOOQ/files/10523347/jOOQ-mcve-kotlin-14564.zip)

**Comment by lukaseder** [comment]
Thanks a lot for the report and MCVE. I can see how this can be a problem.

Note that you probably don't need the generated `equals()` method, as `data classes` already take care of this, sufficiently? See also https://github.com/jOOQ/jOOQ/issues/10917 (which is about `toString()`). The main reason why `equals()` and `hashCode()` may still be useful on generated code is when you have array types, where kotlin doesn't compare array components as I would have expected.

There's a `<pojosEqualsAndHashCode/>` flag to turn that off, in case you don't have arrays, as a workaround.

**Comment by lukaseder** [comment]
I just realised the flag isn't documented in the manual yet: https://github.com/jOOQ/jOOQ/issues/14565

**Comment by apatrida** [comment]
I was going to file a bug for those being generated at all. thanks for the flag.

**Comment by lukaseder** [comment]
I see, I guess there was a misunderstanding with respect to `==` vs `===` in kotlin. It appears that `null == null` (as well as `null.equals(null)`) doesn't NPE in kotlin. In Scala, the latter throws a NPE, whereas the former .

I "hate" to say it, but that's nice :)

**Comment by lukaseder** [comment]
The generated `hashCode()` method is also affected:

```kotlin
result = prime * result + (if (this.id === null) 0 else this.id.hashCode())
```

We should use `==` instead

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.18.0, 3.17.8 (https://github.com/jOOQ/jOOQ/issues/14589), and 3.16.14 (https://github.com/jOOQ/jOOQ/issues/14590)
